config/src/main/resources/goerli.json

```diff
@@ -20,6 +20,12 @@
         "enode://d4f764a48ec2a8ecf883735776fdefe0a3949eb0ca476bd7bc8d0954a9defe8fea15ae5da7d40b5d2d59ce9524a99daedadf6da6283fca492cc80b53689fb3b3@46.4.99.122:32109",
         "enode://d2b720352e8216c9efc470091aa91ddafc53e222b32780f505c817ceef69e01d5b0b0797b69db254c586f493872352f5a022b4d8479a00fc92ec55f9ad46a27e@88.99.70.182:30303"
       ]
+    },
+    "checkpoint": {
+      "hash": "0x2ae30061bdfc7f6dad5b07361dce436502eb0fde68645de12bae4929be619188",
+      "number": 6720000,
+      "totalDifficulty": "0x967F81",
+      "_comment": "must be the beginning of an epoch"
```

Comment:
what happens if this is not a new epoch?  does the difficulty adjustment algorithm break?  clique vote flushing happens at the wrong time, etc?

Reply:
it's because this part of code need to run into an epoch block https://github.com/hyperledger/besu/blob/2abc7074bb54ed0c92ad143dfc96166c1f6ac5a4/consensus/common/src/main/java/org/hyperledger/besu/consensus/common/validator/blockbased/VoteTallyCache.java#L87. if we are not starting on a new epoch we will have the NoSuchElementException 

to avoid changing too much code I prefer to advise starting from a new epoch for clique and like that it will work without problems